Dockerfile

```diff
@@ -0,0 +1,82 @@
+FROM hexpm/elixir:1.15.7-erlang-26.1-debian-bookworm-20230612-slim as base
+
+# install build dependencies
+# --allow-releaseinfo-change allows to pull from 'oldstable'
+RUN apt-get update --allow-releaseinfo-change -y && \
+    apt-get install -y \
+    build-essential \
+    git \
+    openssl \
+    ca-certificates \
+    inotify-tools && \
+    apt-get clean && \
+    rm -f /var/lib/apt/lists/*_*
+
+# Install hex
+RUN mix local.hex --force && \
+    mix local.rebar --force && \
+    mix hex.info
+
+WORKDIR /src
+
+FROM base as deps
+# Cache elixir deps
+ADD mix.exs mix.lock ./
+RUN mix do deps.get, deps.compile
```

Comment:
I'm not sure if it is a good practice to run the `deps.compile` step without setting an environment.

This means that we are budiling test deps while aiming production, but that might even be fine. However, does the dependencies code have different code paths according to the environment?

For the records here is what we were doing:
```dockerfile
# Pass --build-arg BUILD_ENV=dev to build a dev image
ARG BUILD_ENV=prod

ENV MIX_ENV=$BUILD_ENV
```

Here is the default Phoenix generated dockerfile:
```dockerfile
# set build ENV
ENV MIX_ENV="prod"

# install mix dependencies
COPY mix.exs mix.lock ./
RUN mix deps.get --only $MIX_ENV
```

I'm not saying that what we do is wrong, but I'm not 100% sure it is right. We might ask for feedbacks on the elixir forum (https://elixirforum.com/).

Reply:
Fixed, I've added the default `prod' and `docker-compose.dev.yml' to set `MIX_ENV=dev'.